app/header.tsx

```diff
@@ -0,0 +1,84 @@
+'use client'
+
+import Image from 'next/image'
+import Button from './components/button/button'
+import Link from 'next/link'
+import { Navbar } from './components/navbar/navbar'
+import { FC, useEffect, useRef, useState } from 'react'
+import { usePathname } from 'next/navigation'
+
+export const Header: FC<{ s: Record<string, string> }> = ({ s }) => {
+  const [offset, setOffset] = useState(0)
+  const pathname = usePathname()
+
+  useEffect(() => {
```

Comment:
maybe change order and do useState before useEffects?

Reply:
in this one we grouping side effects by it's owned state variables, so it's more clear for identify what own's each other =)